ssz-rs/src/merkleization/mod.rs

```diff
@@ -21,15 +19,35 @@ pub trait Merkleized {
     fn hash_tree_root(&mut self) -> Result<Node, MerkleizationError>;
 }
 
-#[derive(Error, Debug)]
-#[error("the value could not be merkleized: {0}")]
+#[derive(Debug)]
 pub enum MerkleizationError {
-    #[error("failed to serialize value: {0}")]
-    SerializationError(#[from] SerializeError),
-    #[error("cannot merkleize a partial chunk of length {1} (data: {0:?})")]
-    PartialChunk(Vec<u8>, usize),
-    #[error("cannot merkleize data that exceeds the declared limit {0}")]
-    InputExceedsLimit(usize),
+    SerializationError,           // failed to serialize value
+    PartialChunk(Vec<u8>, usize), // cannot merkleize a partial chunk of length {1} (data: {0:?})
+    InputExceedsLimit(usize),     // cannot merkleize data that exceeds the declared limit {0}
+}
+
+impl From<SerializeError> for MerkleizationError {
+    fn from(_: SerializeError) -> Self {
+        return MerkleizationError::SerializationError
+    }
```

Comment:
can we not wrap the serialization error? even if we have to string-ify or something?

Reply:
Sure, good idea! Addressed in https://github.com/ralexstokes/ssz-rs/pull/25/commits/438a003498f6a5be024b30a8137c8673eb05895a.